viewer/src/utilities/RxCounter.ts

```diff
@@ -0,0 +1,52 @@
+/*!
+ * Copyright 2020 Cognite AS
+ */
+
+import { Subject, MonoTypeOperatorFunction, Observable } from 'rxjs';
+import { scan, tap, finalize, shareReplay } from 'rxjs/operators';
+
+enum Action {
+  Add,
+  Remove,
+  Clear
+}
+
+export class RxCounter {
+  private readonly _actionSubject: Subject<Action> = new Subject();
+
+  public incrementOnNext<T>(): MonoTypeOperatorFunction<T> {
+    return tap<T>({
+      next: () => this._actionSubject.next(Action.Add)
+    });
+  }
+
+  public decrementOnNext<T>(): MonoTypeOperatorFunction<T> {
+    return tap<T>({
+      next: () => this._actionSubject.next(Action.Remove)
+    });
+  }
+
+  public resetOnComplete<T>(): MonoTypeOperatorFunction<T> {
+    return finalize<T>(() => {
+      this._actionSubject.next(Action.Clear);
+    });
+  }
```

Comment:
Doc what is meant by `onNext` and `onComplete` (because I don't understand)

Reply:
onNext and onComplete are rx terms. This is an rx operation counter, say if you have a pipeline with events:
-----a-----b-----c-----d-----e
if you add the incrementOnNext then on each submission you will increment.
onComplete will be called once the observable completes.

Reply:
So the way this work, imagine you have something like this.

```
const rxCounter = new RxCounter();
...

const sector$: Observable<Sector> = this.getSectorObservable();
return sector$.pipe(
  rxCounter.incrementOnNext(),
  this.getSectorData(),
  rxCounter.decrementOnNext(),
  rxCounter.resetOnComplete()
);

each sector will increase the counter by 1, each sector with data returned will decrease the counter by 1, if the observable completes the counter is reset
```